gio/src/desktop_app_info.rs

```diff
@@ -48,21 +48,16 @@ pub trait DesktopAppInfoExtManual: IsA<DesktopAppInfo> {
     #[cfg(all(feature = "v2_58", unix))]
     #[cfg_attr(docsrs, doc(cfg(all(feature = "v2_58", unix))))]
     #[doc(alias = "g_desktop_app_info_launch_uris_as_manager_with_fds")]
-    fn launch_uris_as_manager_with_fds<
-        P: IsA<AppLaunchContext>,
-        T: AsRawFd,
-        U: AsRawFd,
-        V: AsRawFd,
-    >(
+    fn launch_uris_as_manager_with_fds<P: IsA<AppLaunchContext>, T: AsFd, U: AsFd, V: AsFd>(
         &self,
         uris: &[&str],
         launch_context: Option<&P>,
         spawn_flags: glib::SpawnFlags,
         user_setup: Option<Box_<dyn FnOnce() + 'static>>,
         pid_callback: Option<&mut dyn (FnMut(&DesktopAppInfo, glib::Pid))>,
-        stdin_fd: &mut T,
-        stdout_fd: &mut U,
-        stderr_fd: &mut V,
+        stdin_fd: T,
+        stdout_fd: U,
+        stderr_fd: V,
```

Comment:
This should be `Option`s (and `None` maps to -1, see the docs)

Reply:
Also `T`, `U`, `V` should probably just become `Option<impl AsFd>` to reduce the syntax noise here